### PR TITLE
Add support for reading INI files

### PR DESCRIPTION
**Pull request by antevens**

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -1682,8 +1684,36 @@ function strip_space {
     echo -n "${@}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
 }
 
-# Usage
-# This function is used to safely edit files for config parameters, etc
+# Load ini file parameter
+# Requires at least two arguments but optionally accepts a third, ini_section
+# If ini_section is specified and multiple sections match an error will be
+# raised. Multiple matches will otherwise be returned
```
Comment (hrpatel) [inline shtdlib.sh]: `s/matches/parameter matches/`?

File: shtdlib.sh [hunk 2]
```diff
@@ -1682,8 +1684,36 @@ function strip_space {
...
+
+    if [ -n "${ini_section}" ]; then
+        #shellcheck disable=SC2086
+        ini_section_match="$(grep -c "\[${ini_section}\]" "${filename}")"
+        if [ "${ini_section_match}" -lt 1 ]; then
+            color_echo red "Unable to find INI section matching ${ini_section}"
+            return 1
+        elif [ "${ini_section_match}" -eq 1 ]; then
+            debug 9 "Found INI section ${ini_section}"
+            sed -n "/\[${ini_section}\]/,/\[/p" "${filename}" | grep -E "^${name}" | awk -F= '{print $2}'
```
Comment (hrpatel) [inline shtdlib.sh]: Why not limit the grep search to just 1 result?

File: shtdlib.sh (hunk 1 above)
Comment (hrpatel) [reply shtdlib.sh]: Unless this sentence is related to the one before about multiple INI sections

Comment (antevens) [reply shtdlib.sh]: Clarified

File: shtdlib.sh (hunk 2 above)
Comment (antevens) [reply shtdlib.sh]: good point

**Pull request merged**